Atomic/rpmwriter.py

```diff
@@ -0,0 +1,369 @@
+import os
+import sys
+import struct
+import subprocess
+import select
+import gzip
+import hashlib
+import tempfile
+import shutil
+
+# More information on the RPM file format can be found here:
+# http://ftp.rpm.org/max-rpm/s1-rpm-file-format-rpm-file-format.html
+# The same documentation was used to initially write this module.
+#
+# Only tags that are actually used in the code were added to the RpmWriter class.
+# The complete list is available here:
+# https://github.com/rpm-software-management/rpm/blob/master/lib/rpmtag.h
+
+class RpmWriter(object):
```

Comment:
We're not *building* here though.  More like just repacking binaries.  What other use cases are there where one isn't using a spec file?  Or without wanting a dependency on `rpmbuild`?